File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/annotation/RequirePermission.java
package com.nivasafinance.features.rolemanagement.annotation;

import com.nivasafinance.features.rolemanagement.enums.ActionEnum;
import com.nivasafinance.features.rolemanagement.enums.ModuleEnum;
import com.nivasafinance.features.rolemanagement.enums.OperationsEnum;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface RequirePermission {
    ActionEnum action();
    ModuleEnum module();
    OperationsEnum operation();
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/annotation/RequirePermission.kt
package com.nivasafinance.features.rolemanagement.annotation

import com.nivasafinance.features.rolemanagement.enums.ActionEnum
import com.nivasafinance.features.rolemanagement.enums.ModuleEnum
import com.nivasafinance.features.rolemanagement.enums.OperationsEnum

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class RequirePermission(
    val action: ActionEnum,
    val module: ModuleEnum,
    val operation: OperationsEnum
)
